expression/substring.go

```diff
@@ -202,3 +203,83 @@ func (f *FunctionSubstringIndex) Eval(ctx context.Context, args map[interface{}]
 func (f *FunctionSubstringIndex) Accept(v Visitor) (Expression, error) {
 	return v.VisitFunctionSubstringIndex(f)
 }
+
+// FunctionLocate returns the position of the first occurrence of substring.
+// See: https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_locate
+type FunctionLocate struct {
+	Str    Expression
+	SubStr Expression
+	Pos    Expression
+}
+
+// Clone implements the Expression Clone interface.
+func (f *FunctionLocate) Clone() Expression {
+	nf := &FunctionLocate{
+		Str:    f.Str.Clone(),
+		SubStr: f.SubStr.Clone(),
+	}
+	if f.Pos != nil {
+		nf.Pos = f.Pos.Clone()
+	}
+	return nf
+}
+
+// IsStatic implements the Expression IsStatic interface.
+func (f *FunctionLocate) IsStatic() bool {
+	return f.Str.IsStatic() && f.SubStr.IsStatic() && (f.Pos == nil || f.Pos.IsStatic())
+}
+
+// String implements the Expression String interface.
+func (f *FunctionLocate) String() string {
+	if f.Pos != nil {
+		return fmt.Sprintf("LOCATE(%s, %s, %s)", f.SubStr, f.Str, f.Pos)
+	}
+	return fmt.Sprintf("LOCATE(%s, %s)", f.SubStr, f.Str)
+}
+
+// Eval implements the Expression Eval interface.
+func (f *FunctionLocate) Eval(ctx context.Context, args map[interface{}]interface{}) (interface{}, error) {
+	// eval str
+	fs, err := f.Str.Eval(ctx, args)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	str, ok := fs.(string)
```

Comment:
Seems MySQL support some none-string args like `SELECT LOCATE(1, 2)`? 
Any need to convert fs to string?